docs/fundamentals/environments.rst

```diff
@@ -1,8 +1,127 @@
+	:language: c#
+	:caption: Startup.cs (some parts removed for brevity)
+	:linenos:
+	:dedent: 4
+	:lines: 27-47,89-109,134-135
+	:emphasize-lines: 8,10-15,30-41
+
+The highlighted sections in the example above show several examples of adjusting application configuration based on environment.
+
+In the ``Startup()`` method (constructor), the configuration of the application is set up to optionally allow for environment-specific configuration files (e.g. ``config.Development.json``) that will override any other config settings (because the file is added last in the configuration setup chain - :doc:`learn more about ASP.NET configuration <configuration>`). A call to ``IsEnvironment()`` is also used to ensure that User Secrets are only configured for use in ``Development`` (learn more about User Secrets and the `Secret Manager <https://github.com/aspnet/Home/wiki/DNX-Secret-Configuration>`_).
+
+In ``Configure()``, the environment is checked once more, and if the app is running in a ``Development`` environment, then it enables BrowserLink and error pages (which typically should not be run in production). Otherwise, if the app is not running in a development environment, a standard error handling page is configured to be displayed in response to any unhandled exceptions.
+
+Environment variables
+---------------------
+
+In addition to using Visual Studio's project properties and ``launchSettings.json`` file to set environment variables, you can of course manage environment variables yourself from the command line. On Windows, you can use ``set`` to view current environment variables, and ``set ASPNET_ENV=Development`` to set the current environment to development. You can filter the list by piping (using the ``|`` character) the result to ``findstr``. The following screenshot shows how to view the current ASPNET_ENV setting (if any), how to set it, and then how to run the sample application from the command prompt:
+
+.. image:: environments/_static/windows-command-environment.png
+
+On a Mac, you can do the same thing, using a slightly different set of commands. Open a Terminal window, and use ``export`` to see a list of currently configured environment variables. You can filter the result by piping it to ``grep``. Use ``export ASPNET_ENV=Development`` to set the variable, as shown.
+
+.. image:: environments/_static/mac-terminal-environment.png
+
+At this point you could launch the application, perhaps using ``Kestrel`` as your web server. :doc:`Learn more about developing ASP.NET applications on a Mac <../tutorials/your-first-mac-aspnet>`.
+
+Summary
+-------
+
+ASP.NET 5 leverages the ``ASPNET_ENV`` environment variable to control how the application behaves in different environments. When publishing an application from development to staging to production, setting this variable appropriately for the environment allows for optimization of the application for debugging, testing, or production use, as appropriate.
```

Comment:
Instead of focusing on the specific env var can we have the summary focus on the fact that ASP.NET 5 provides a first-class notion of the application environment? The env var is just the current mechanism we use for specifying the environment - the important thing here is the environment abstraction that you can use in your app.